jmespath-jakarta-jsonp/src/main/java/io/burt/jmespath/jakarta/jsonp/JsonpRuntime.java

```diff
@@ -0,0 +1,230 @@
+      case STRING:
+        return ((JsonString)value).getString().length() > 0;
+      default:
+        throw new IllegalStateException(String.format("Unknown node type encountered: %s", value.getValueType()));
+    }
+  }
+
+  @Override
+  public JmesPathType typeOf(JsonValue value) {
+    switch (value.getValueType()) {
+      case ARRAY:
+        return JmesPathType.ARRAY;
+      case OBJECT:
+        return JmesPathType.OBJECT;
+      case STRING:
+        return JmesPathType.STRING;
+      case FALSE:
+      case TRUE:
+        return JmesPathType.BOOLEAN;
+      case NULL:
+        return JmesPathType.NULL;
+      case NUMBER:
+        return JmesPathType.NUMBER;
+      default:
+        throw new IllegalStateException(String.format("Unknown node type encountered: %s", value.getValueType()));
+    }
+  }
+
+  @Override
+  public JsonValue getProperty(JsonValue value, JsonValue name) {
+    if (value.getValueType() == OBJECT) {
+      return nodeOrNullNode(((JsonObject)value).get(textOnStringOrNull(name)));
+    } else {
+      return JsonValue.NULL;
+    }
+  }
+
+  @Override
+  public Collection<JsonValue> getPropertyNames(JsonValue value) {
+    if (value.getValueType() == OBJECT) {
+      Set<String> nameSet = ((JsonObject)value).keySet();
+      List<JsonValue> names = new ArrayList<>(nameSet.size());
+      for(String n : nameSet) {
+        names.add(createString(n));
+      }
+      return names;
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  @Override
+  public JsonValue createNull() {
+    return nodeOrNullNode(null);
+  }
+
+  @Override
+  public JsonValue createArray(Collection<JsonValue> elements) {
+    JsonArrayBuilder builder = Json.createArrayBuilder();
+    for(JsonValue element : elements) {
+      builder.add(useJsonNull(element));
+    }
+    return builder.build();
+  }
+
+  @Override
+  public JsonValue createString(String str) {
+    return useJsonNull(Json.createValue(str));
+  }
+
+  @Override
+  public JsonValue createBoolean(boolean b) {
+    return b ? JsonValue.TRUE : JsonValue.FALSE;
+  }
+
+  @Override
+  public JsonValue createObject(Map<JsonValue, JsonValue> obj) {
+    JsonObjectBuilder builder = Json.createObjectBuilder();
+    for (Map.Entry<JsonValue, JsonValue> entry : obj.entrySet()) {
+      String key = textOnStringOrNull(entry.getKey());
+      if (key != null) {
+        builder.add(key, useJsonNull(entry.getValue()));
+      }
+    }
+    return builder.build();
+  }
+
+  @Override
+  public JsonValue createNumber(double n) {
+    return Json.createValue(n);
+  }
+
+  @Override
+  public JsonValue createNumber(long n) {
+    return Json.createValue(n);
+  }
+
+  private JsonValue useJsonNull(JsonValue value) { return (value == null) ? JsonValue.NULL : value; }
+
+  private JsonValue nodeOrNullNode(JsonValue node) {
+    if (node == null) {
+      return JsonValue.NULL;
+    } else {
+      return node;
+    }
+  }
+
+  private String textOnStringOrNull(JsonValue value) {
```

Comment:
This name confuses me, specifically the "on" part. Could it be called `stringOrNull` or `textOrNull`?

Reply:
I've looked at the places where this method is used, and I think it's in fact not needed. The callers (`createObject` and `getProperty`) don't need to call this method.

Reply:
I replaced it by the toString(key) call.

Reply:
(Still being a bit paranoid here.)